clients/js/evm.ts

```diff
@@ -480,3 +486,77 @@ export async function setStorageAt(rpc: string, contractAddress: string, storage
     ],
   })).data
 }
+
+export async function checkMainnetContract(
+  chainId: ChainId,
+  nodeUrl: string | undefined
+) {
+  if (chainId !== CHAIN_ID_ETH && chainId !== CHAIN_ID_BSC) {
+    throw Error(`Invalid evm chain id ${chainId}`)
+  }
+  const config = chainId === CHAIN_ID_ETH ? ethereumMainnetConfig : bscMainnetConfig
+  const provider = new ethers.providers.JsonRpcProvider(nodeUrl ?? config.nodeUrl)
+  await checkGovernance(chainId, config.contracts.governance, provider)
+  await checkTokenBridge(chainId, config.contracts.tokenBridge, config.contracts.wrappedNative, provider)
+  console.log(`The deployed contract is correct`)
+}
+
+async function checkGovernance(
+  expectedChainId: ChainId,
+  governanceAddress: string,
+  provider: ethers.providers.JsonRpcProvider
+) {
+  const governance = ethers_contracts.Implementation__factory.connect(governanceAddress, provider)
+  const index = await governance.getCurrentGuardianSetIndex()
+  const guardianSet = await governance.getGuardianSet(index)
+  const signers = guardianSet[0]
+  signers.forEach((address, index) => {
+    const expected = guardianMainnetConfig.initSigners[index]
+    if (expected.toLowerCase() !== address.toLowerCase()) {
+      throw Error(`Invaid guardian signer at index ${index}, expected ${expected}, got ${address}`)
+    }
+  })
+
+  checkChainParams(
+    (await governance.chainId()) as ChainId,
+    await governance.governanceChainId(),
+    await governance.governanceContract(),
+    expectedChainId
+  )
+}
+
+async function checkTokenBridge(
+  expectedChainId: ChainId,
+  tokenBridgeAddress: string,
+  expectedWrappedNative: string,
+  provider: ethers.providers.JsonRpcProvider
+) {
+  const tokenBridge = ethers_contracts.BridgeImplementation__factory.connect(tokenBridgeAddress, provider)
+  checkChainParams(
+    (await tokenBridge.chainId()) as ChainId,
+    await tokenBridge.governanceChainId(),
+    await tokenBridge.governanceContract(),
+    expectedChainId
+  )
+  const wrappedNative = await tokenBridge.WETH()
```

Comment:
Q: Dose `WETH` return `WBNB` for BSC chain?

Reply:
Yes, in the wormhole contract, `WETH` is used to represent the wrapped native address on EVM chains.